openapi/src/parts/workspace.yaml

```diff
@@ -202,6 +202,37 @@ paths:
         '500':
           $ref: '#/components/responses/ServerError'
 
+  /api/workspaces/v1/{workspaceId}/policies/link:
```

Comment:
Could you make a one pager explaining what this API is for? It is not obvious to me.
In general, I prefer having new APIs written down and looked over before making their appearance in the YAML...

Having read more of the code, I am don't think this is the right level of abstraction. If a workspace should inherit a policy from a billing profile or from a landing zone, then we should build that in WSM. Not build an open ended interface at the policy level that allows any workspace writer to tinker with the policy database.

Reply:
this is for linking a workspace's policy to a snapshot's. It is essentially a wrapper around https://tps.dsde-dev.broadinstitute.org/#/Tps/linkPao that verifies everything complies first.

Reply:
If we made the connection to the snapshot a referenced resource, then this policy would flow in the same way as we do for data collections and other workspaces. I don't think we want the naked link as the API here.

Reply:
here are some docs https://docs.google.com/document/d/1AtnBFMw5YhxWNMVeXRN5g9Z8JBKxbfp1iNnD8IsKdZY/edit#heading=h.90m9ste9j9qq

I am not 100% sure modeling this as a reference makes sense. We are actually copying data into a workspace, not referencing anything.

How does this work today for copying data from other workspaces?

Reply:
This works as part of the clone operation. Whether you are making a reference or copying the data, the workspace inherits the policy of the source. Maybe we need a variant of clone that has a snapshot as a source. Presumably, you could make a reference or a controlled resource as the target based on cloning instructions.

Alternately, some policy-aware data movement component that can interact with TPS to configure the policies. But that seems like too big a lift.